### PR TITLE
Feature: named parts

### PR DESCRIPTION
**Pull request by Bouni**
This PR adds the ability to pass names along with the lengths which are included in the results as well.

Here's example data:
```json
{"cut_width": "3",
 "max_length": "1500",
 "target_sizes": [{"length": "200", "name": "Part1", "quantity": "4"},
                  {"length": "500", "quantity": "2"}]}
```

The resulting data looks like this:
```json
{"job": {"cut_width": 3,
         "max_length": 1500,
         "target_sizes": [{"length": 200, "name": "Part1", "quantity": 4},
                          {"length": 500, "name": "", "quantity": 2}]},
 "lengths": [[[500, ""], [500, ""], [200, "Part1"], [200, "Part1"]],
             [[200, "Part1"], [200, "Part1"]]],
 "solver_type": "bruteforce",
 "time_us": 4974}
```

This will allow the Frontend to show part names along with the dimensions which makes it much easier to find parts in the cutlist.

> [!CAUTION]
> This is a breaking change and the Frontend must be updated as well. I'll submit a PR for that soon

### CODE REVIEW

**Comment by Bouni** [comment]
I just realized that the output is not rigtht (for whatever reason, the tests seem to check for the right data 🤔)

**Comment by Bouni** [comment]
False alarm 🤦🏽 I had a stock length of 13000 in my example which resulted in just one element in the resulting array.

**Comment by ModischFabrications** [comment]
Thanks for contributing 👍🏻
I will wait until your PR for the frontend is done, feel free to ask if anything is unclear

**Comment by Bouni** [comment]
I realized a issue yesterday. I have not refactored the FFD solver yet and if a user adds to many parts it will not work.
I'll look into it today. The frontend is already done, you can check it out here: https://cutsolver.bouni.de/

**Comment by Bouni** [comment]
@ModischFabrications I refactored the FFD solver as well, all tests pass and my test setup works like a charm!

grafik

**Comment by Bouni** [comment]
One last question, in the comment above the ffd solver is a TODO comment about the execution time being aweful. what size of test data did you use?

**Comment by ModischFabrications** [comment]
> One last question, in the comment above the ffd solver is a [TODO comment](https://github.com/ModischFabrications/CutSolver/blob/master/app/solver/solver.py#L142) about the execution time being aweful. what size of test data did you use?

Not sure, it's been a while. There is a benchmarking-test somewhere in there, but I haven't set up the tests on my end again.

**Comment by ModischFabrications** [comment]
@Bouni tests failed, it seems that you include py 3.11 in your dependencies but chose a py 3.12 docker image. Can you fix it, so that the CI runs through successfully? Everything else seems fine at first glance, I will merge und update the service once both PRs are done

**Comment by Bouni** [comment]
I'll do that tomorrow. I think it fails because of a python 3.9 in the ci.
For me everything runs fine in a python 3.12 container

File: Dockerfile
Comment (ModischFabrications) [inline Dockerfile:2]:
```suggestion
FROM python:3.11 as build
```

File: Dockerfile
Comment (ModischFabrications) [inline Dockerfile:12]:
```suggestion
FROM python:3.11-slim
```

**Review by ModischFabrications — CHANGES_REQUESTED** [review]
I guess this could be it, CI pipenv is rather picky with it's dependencies.

**Comment by Bouni** [comment]
This is what I mean

https://github.com/ModischFabrications/CutSolver/blob/master/.github%2Fworkflows%2Fci.yml#L42

**Review by ModischFabrications — APPROVED** [review]

**Comment by ModischFabrications** [comment]
Good call, everything works now. Blocked by https://github.com/ModischFabrications/CutSolverFrontend/pull/63

**Comment by Bouni** [comment]
I guess we might shoul update the python version to 3.12 as that's the current stabel release, what do you think?

**Comment by ModischFabrications** [comment]
I don't see the need, but I'm open to it. Feel free to update if you prefer.

**Comment by Bouni** [comment]
You're right I think we just leave it at 3.11 and update it maybe at a later point

**Comment by ModischFabrications** [comment]
FYI: CutSolver (Backend) was promoted to v1.0.0 and got it's releases: https://hub.docker.com/r/modischfabrications/cutsolver/tags

Thanks for your support!

**Pull request merged**
